src/Functions/getSetting.cpp

```diff
@@ -58,17 +61,51 @@ class FunctionGetSetting : public IFunction, WithContext
             throw Exception(ErrorCodes::ILLEGAL_COLUMN,
                             "The argument of function {} should be a constant string with the name of a setting",
                             String{name});
+        if (arguments.size() == 2)
+        {
+            const auto * default_value_column = arguments[1].column.get();
+            if (!default_value_column || default_value_column->size() != 1)
+            {
+                throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT,
+                                "The 2nd argument of function {} should be a constant", String{name});
+            }
+        }
 
         std::string_view setting_name{column->getDataAt(0).toView()};
-        return getContext()->getSettingsRef().get(setting_name);
+        Field setting_value;
+        try
+        {
+            setting_value = getContext()->getSettingsRef().get(setting_name);
+        }
+        catch(...)
+        {
+            if (error_handling == ErrorHandling::Exception)
+                throw;
+            else if (error_handling == ErrorHandling::Default)
+            {
+                auto default_value_column = arguments[1].column;
+
+                setting_value = (*default_value_column)[0];
+            }
+        }
```

Comment:
Use `tryGet(setting_name, field)` instead

Reply:
If I use tryGet(), then specific exception will have to be thrown from here.  That will lead to hard dependency on SettingNotFound() exception. The Settings implementation could change tomorrow to throw new types of exception. Let me know your opinion!

Reply:
yes, but you could use `get()` if it's `Exception` mode, and `tryGet()` otherwise